public/pages/QueryInsights/QueryInsights.tsx

```diff
@@ -88,56 +160,85 @@ const QueryInsights = ({
+        const result = calculateMetric(
+          measurements?.latency?.number,
+          measurements?.latency?.count,
+          1,
+          METRIC_DEFAULT_MSG
+        );
+        return `${result} ms`;
       },
       sortable: true,
       truncateText: true,
     },
     {
       field: MEASUREMENTS_FIELD,
       name: CPU_TIME,
-      render: (measurements: { cpu?: { number?: number } }) => {
-        const cpuValue = measurements?.cpu?.number;
-        return cpuValue !== undefined ? `${cpuValue / 1000000} ms` : METRIC_DEFAULT_MSG;
+      render: (measurements: SearchQueryRecord['measurements']) => {
+        const result = calculateMetric(
+          measurements?.cpu?.number,
+          measurements?.cpu?.count,
+          1000000, // Divide by 1,000,000 for CPU time
+          METRIC_DEFAULT_MSG
+        );
+        return `${result} ms`;
       },
       sortable: true,
       truncateText: true,
     },
     {
       field: MEASUREMENTS_FIELD,
       name: MEMORY_USAGE,
-      render: (measurements: { memory?: { number?: number } }) => {
-        const memoryValue = measurements?.memory?.number;
-        return memoryValue !== undefined ? `${memoryValue} B` : METRIC_DEFAULT_MSG;
+      render: (measurements: SearchQueryRecord['measurements']) => {
+        const result = calculateMetric(
+          measurements?.memory?.number,
+          measurements?.memory?.count,
+          1,
+          METRIC_DEFAULT_MSG
+        );
+        return `${result} B`;
       },
       sortable: true,
       truncateText: true,
     },
     {
       field: INDICES_FIELD,
       name: INDICES,
-      render: (indices: string[]) => Array.from(new Set(indices.flat())).join(', '),
+      render: (indices: string[], query: SearchQueryRecord) => {
+        const isSimilarity = query.group_by === 'SIMILARITY';
+        return isSimilarity ? '-' : Array.from(new Set(indices.flat())).join(', ');
+      },
       sortable: true,
       truncateText: true,
     },
     {
       field: SEARCH_TYPE_FIELD,
       name: SEARCH_TYPE,
-      render: (searchType: string) => searchType.replaceAll('_', ' '),
+      render: (searchType: string, query: SearchQueryRecord) => {
+        const isSimilarity = query.group_by === 'SIMILARITY';
+        return isSimilarity ? '-' : searchType.replaceAll('_', ' ');
+      },
       sortable: true,
       truncateText: true,
     },
     {
       field: NODE_ID_FIELD,
       name: NODE_ID,
+      render: (nodeId: string, query: SearchQueryRecord) => {
+        const isSimilarity = query.group_by === 'SIMILARITY';
+        return isSimilarity ? '-' : nodeId;
+      },
       sortable: true,
       truncateText: true,
     },
     {
       field: TOTAL_SHARDS_FIELD,
       name: TOTAL_SHARDS,
+      render: (totalShards: number, query: SearchQueryRecord) => {
+        const isSimilarity = query.group_by === 'SIMILARITY';
+        return isSimilarity ? '-' : totalShards;
+      },
```

Comment:
Some of these don't seem to make sense if looking at group_by. Should we remove some columns from the table depending on the type you are looking at? Which brings up another question, I wasn't able to get to a situation where both group and single queries show up. If it is only one or the other, IMO it is confusing to put type filter - it should just be one or the other based on your configuration.